src/MCPClient/lib/clientScripts/restructureDIPForContentDMUpload.py

```diff
@@ -22,614 +22,118 @@
-# archivematicaCommon
-from archivematicaFunctions import normalizeNonDcElementName
-from executeOrRunSubProcess import executeOrRun
+import os
+import re
+import sys
+
+import archivematicaXMLNamesSpace as ns
 
-# Create the output dir for the CONTENTdm DIP and return the resulting path.
-# importMethod is either 'projectclient' or 'directupload'.
-def prepareOutputDir(outputDipDir, importMethod, dipUuid):
-    outputDipDir = os.path.join(outputDipDir, 'CONTENTdm', importMethod, dipUuid)
-    # Check for and then delete a subdirectory named after the current package. We always want
-    # a clean output directory for the import package.
-    if os.path.exists(outputDipDir):
-        shutil.rmtree(outputDipDir)
-    os.makedirs(outputDipDir)
-    return outputDipDir
+# archivematicaCommon
+import archivematicaFunctions
 
+def parseDmdSec(dmdSec, label='[Placeholder title]'):
+    """
+    Parses a dmdSec into a dict with child tag names and their values
 
-# Takes in a DOM object containing the DC or OTHER dmdSec, returns a dictionary with 
-# tag : [ value1, value2] members. Also, since minidom only handles byte strings
-# so we need to encode strings before passing them to minidom functions. 'label' is
-# an optional arguement for use with compound item children.
-def parseDmdSec(dmdSec, label = '[Placeholder title]'):
+    :param dmdSec: dmdSec elements
+    :param label: Default title if not provided. Required by CONTENTdm
+    :returns: Dict of {<child element tag>: [<value>, ...]
+    """
     # If the dmdSec object is empty (i.e, no DC metadata has been assigned
     # in the dashboard, and there was no metadata.csv or other metadata file
     # in the transfer), return a placeholder title.
     if dmdSec is None:
-        return {'title' : [label]}
-    if not hasattr(dmdSec, 'getElementsByTagName'):
-        return {'title' : [label]}    
+        return {'title': [label]}
+    elementsDict = collections.OrderedDict()
 
-    mdWraps = dmdSec.getElementsByTagName('mdWrap')
-    mdType = mdWraps[0].attributes["MDTYPE"]
-    
     # If we are dealing with a DOM object representing the Dublin Core metadata,
-    # check to see if there is a title (required by CONTENTdm). If not, assign a 
+    # check to see if there is a title (required by CONTENTdm). If not, assign a
     # placeholder title and return.
-    if mdType == 'DC' and hasattr(dmdSec, 'getElementsByTagName'):
-        dcTitlesDom = dmdSec.getElementsByTagName('title')
+    mdType = dmdSec.xpath('mets:mdWrap/@MDTYPE', namespaces=ns.NSMAP)
+    if mdType == 'DC':
+        dcTitlesDom = dmdSec.findall('.//dcterms:title', namespaces=ns.NSMAP)
         if not dcTitlesDom:
-            return {'title' : '[Placeholder title]'} 
+            elementsDict['title'] = label
+
+    # Iterate over all descendants and put in the return dict
+    # Key is the element's tag name, value is a list of the element's text
+    # TODO check for problems with encoding & unicode vs str
```

Comment:
Checked for problems and miraculously didn't find any. If the text is unicode it writes out just fine.